src/main/scala/gitbucket/core/controller/RepositoryViewerController.scala

```diff
@@ -675,15 +675,16 @@ trait RepositoryViewerControllerBase extends ControllerBase {
       val oid = git.getRepository.resolve(revision)
       val revCommit = JGitUtil.getRevCommitFromId(git, oid)
       val sha1 = oid.getName()     
-      val filename = repository.name + "-" +
-        (if(sha1.startsWith(revision)) sha1 else revision).replace('/','-') + suffix
+      val repositorySuffix = (if(sha1.startsWith(revision)) sha1 else revision).replace('/','-')
+      val filename = repository.name + "-" + repositorySuffix + suffix
 
       contentType = "application/octet-stream"
       response.setHeader("Content-Disposition", s"attachment; filename=${filename}")
       response.setBufferSize(1024 * 1024);
 
       git.archive
          .setFormat(suffix.tail)
+         .setPrefix(repository.name + "-" + repositorySuffix + "/")
```

Comment:
Is the modification in this file same as #1547? If true, I hope you remove the commit from this pull request.